core/src/main/scala/org/apache/spark/deploy/worker/Worker.scala

```diff
@@ -362,11 +362,13 @@ private[spark] class Worker(
             self,
             workerId,
             host,
+            webUiPort,
```

Comment:
I'm working on a PR now that will fix this and an issue where this doesn't obey SPARK_PUBLIC_DNS (the latter is a bit trickier to test, unfortunately, since it needs some tricky SparkConf + Mockito usage to mock environment variables).